### PR TITLE
Add dockerfiles and docker-compose config

### LINKED ISSUE #6: Add Dockerfiles for front/backend and create a docker-compose config to run the application
Depends on #5 and #4 to be merged.

Currently there is no simple way to run the entire application and Dockerfiles are needed for front and backend. Also a docker-compose file that includes those two images and mongodb would be convenient.

Regarding the setup I see two options:
- Somehow include the frontend code in the backend Dockerimage and use the express server to serve the frontend
- Setup a small nginx reverse proxy to route frontend requests to a frontend server and backend requests to the backend server
Any preferences/ideas about this?

Btw you can assign me to this issue, will take care of it

### PR DESCRIPTION
**Pull request by steve192**
Closes #6

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    image: mongo:latest
+    ports:
+      - 27017:27017
+  proxy:
+    image: nginx:stable-alpine
+    environment:
+      - NGINX_ENVSUBST_TEMPLATE_SUFFIX=.conf
+      - NGINX_ENVSUBST_OUTPUT_DIR=/etc/nginx
+    volumes:
+      - ${PWD}/nginx.conf:/etc/nginx/templates/nginx.conf.conf
```
Comment (aerabi) [inline docker-compose.yml:20]: Is it a typo?

```suggestion
      - ${PWD}/nginx.conf:/etc/nginx/templates/nginx.conf
```

**Review by aerabi — APPROVED** [review]
Really awesome! 😎 Copying the static files into the NGINX image was also very smart. I had never thought of it. 😅 
Perhaps there is but one typo.

**Comment by steve192** [comment]
Thanks 😄

Typos ... Not my best strength 😄

File: docker-compose.yml (hunk 1 above)
Comment (steve192) [reply docker-compose.yml:20]: Nope not a typo. The nginx image uses this file as a template file and replaces its config with it (suffix defined in NGINX_ENVSUBST_TEMPLATE_SUFFIX=.conf)

We could change it to .template if that makes it more understandable

**Comment by steve192** [comment]
> Really awesome! 😎 Copying the static files into the NGINX image was also very smart. I had never thought of it. 😅 Perhaps there is but one typo.

It's very convenient if you want to debug parts of your application. You can simply replace the urls in the nginx config with host.docker.internal urls so that requests are routed to the application running in your ide.

That's my go to setup when working with multiple services.

**Pull request merged**